### PR TITLE
Fix failed to commit and rollback the connection while throw sql exception

### PR DESCRIPTION
**Pull request by thushaaanthan**
When we use the JDBC transaction by applying connection.commit() to commit the SQL statements and make sure SQL statements within a transaction block are all executed successfully. if either one of the SQL statement within the transaction block is failed, abort and rollback everything within the transaction block.
In the identity-migration-resources repository, there are some places use the JDBC transaction but did not properly commit and rollback the transaction.

Issue: https://github.com/wso2/product-is/issues/5778

### CODE REVIEW

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v550/migrator/OAuthDataMigrator.java [hunk 1]
```diff
@@ -66,19 +66,28 @@ public void addHashColumns() throws MigrationClientException, SQLException {
...
         if (!isAuthzCodeHashColumnAvailable) {
             try (Connection connection = getDataSource().getConnection()) {
-                AuthzCodeDAO.getInstance().addAuthzCodeHashColumns(connection);
-                connection.commit();
+                connection.setAutoCommit(false);
+                try {
+                    AuthzCodeDAO.getInstance().addAuthzCodeHashColumns(connection);
+                    connection.commit();
+                } catch (SQLException e) {
+                    connection.rollback();
```
Comment (ruwanta) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v550/migrator/OAuthDataMigrator.java]: Need to log error or throw the SQL exception. Here we have eaten the exception, which makes harder to see any errors.

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v540/migrator/PermissionMigrator.java [hunk 2]
```diff
@@ -47,38 +47,41 @@ public void migrate() throws MigrationClientException {
...
-            updateRolePermissionTable(connection, duplicatedPermissions);
-
-            List<UserPermission> duplicatedUserPermissions = getDuplicatedUserPermissions(connection,
-                    duplicatedPermissions);
-            if (!duplicatedUserPermissions.isEmpty()) {
-                log.info(Constant.MIGRATION_LOG + " Found " + duplicatedUserPermissions.size() + " duplicated user "
-                        + "permissions.");
-                deleteDuplicatedUserPermissions(connection, duplicatedUserPermissions);
-                log.info(Constant.MIGRATION_LOG + " Removed duplicated user permissions.");
+            try {
```
Comment (ruwanta) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v540/migrator/PermissionMigrator.java]: Can we retain previous formatting

Comment (thushaaanthan) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v540/migrator/PermissionMigrator.java]: I have added new try, that's why it changed the format

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v550/migrator/OAuthDataMigrator.java (hunk 1 above)
Comment (thushaaanthan) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v550/migrator/OAuthDataMigrator.java]: Added

**Pull request merged**